diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad05ac1..e2c1e87 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,6 +96,7 @@ endif()
 
 target_link_libraries(rmf_traffic
   PUBLIC
+    Eigen3::Eigen
     rmf_utils::rmf_utils
     Threads::Threads
   PRIVATE
diff --git a/src/rmf_traffic/blockade/geometry.cpp b/src/rmf_traffic/blockade/geometry.cpp
index 049351e..d69a999 100644
--- a/src/rmf_traffic/blockade/geometry.cpp
+++ b/src/rmf_traffic/blockade/geometry.cpp
@@ -17,6 +17,7 @@
 
 #include "geometry.hpp"
 
+#include <cassert>
 #include <cmath>
 
 namespace rmf_traffic {
diff --git a/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h b/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h
index 861f72d..a239104 100644
--- a/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h
+++ b/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h
@@ -41,6 +41,8 @@
 #ifndef FCL_CCD_TAYLOR_MODEL_INL_H
 #define FCL_CCD_TAYLOR_MODEL_INL_H
 
+#include <cassert>
+
 #include "fcl/math/motion/taylor_model/taylor_model.h"
 
 namespace fcl
