diff --git a/src/unix_signal_handler.cpp b/src/unix_signal_handler.cpp
index beb2369..48da6f6 100644
--- a/src/unix_signal_handler.cpp
+++ b/src/unix_signal_handler.cpp
@@ -67,7 +67,7 @@ void UnixSignalHandler::setup()
 
   struct sigaction int_action;
   int_action.sa_handler = UnixSignalHandler::intSignalHandler;
-  ::sigemptyset(&int_action.sa_mask);
+  sigemptyset(&int_action.sa_mask);
   int_action.sa_flags = SA_RESTART;
   if (::sigaction(SIGINT, &int_action, nullptr)) {
     qFatal("UnixSignalHandler: couldn't install SIGINT handler");
@@ -75,7 +75,7 @@ void UnixSignalHandler::setup()
 
   struct sigaction term_action;
   term_action.sa_handler = UnixSignalHandler::termSignalHandler;
-  ::sigemptyset(&term_action.sa_mask);
+  sigemptyset(&term_action.sa_mask);
   term_action.sa_flags = SA_RESTART;
   if (::sigaction(SIGTERM, &term_action, nullptr)) {
     qFatal("UnixSignalHandler: couldn't install SIGTERM handler");
