diff --git a/CMakeLists.txt b/CMakeLists.txt
index db6c035..5bb36a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,8 @@ include(CheckLibraryExists)
 
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_ros_core REQUIRED)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
 
 if(UNIX AND NOT APPLE)
   include(cmake/check_c_compiler_uses_glibc.cmake)
@@ -87,8 +89,11 @@ if(BUILD_TESTING AND NOT RCUTILS_DISABLE_FAULT_INJECTION)
 endif()
 
 target_link_libraries(${PROJECT_NAME}
-  ${CMAKE_DL_LIBS}
-  ament_cmake_ros_core::ament_ros_defaults
+  PUBLIC
+    ${CMAKE_DL_LIBS}
+    ament_cmake_ros_core::ament_ros_defaults
+  PRIVATE
+    Threads::Threads
 )
 
 check_library_exists(atomic __atomic_load_8 "" HAVE_LIBATOMICS)
@@ -98,8 +103,9 @@ if(HAVE_LIBATOMICS AND NOT WIN32 AND NOT APPLE)
   # Don't export it - downstream packages get atomic symbols transitively
   # through librcutils.so. See https://github.com/ros2/rcutils/issues/525
   target_link_libraries(${PROJECT_NAME}
-    atomic
-    ament_cmake_ros_core::ament_ros_defaults
+    PUBLIC
+      atomic
+      ament_cmake_ros_core::ament_ros_defaults
   )
 endif()
 
